### DISCUSSION TITLE
automation values discussion/clarification

### DISCUSSION THREAD

**Discussion by turbolooser** [revk/xESP32-Faikout#217] [General]
As I am trying to understand the auto-mode (which is absolutely awesome with the external BLE temp @revk ) there are some numbers I do not quite understand and would like to start a discussion around.

When checking the automation topic on mqtt I do see this faikin-wohnzimmer like:
{"hot":true,"approaching":1800,"beyond":0,"samples":1800,"period":900,"temp":17.81,"min":19.10,"max":20.50}

webgui:
<img width="448" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/3786208a-531f-4fd0-bf3b-f238cc58a7f9">

I cannot get these numbers correctly to my expectations.
My target temp is set to 19.5
My expectation would be that given the +-0.5 setting in auto-mode:
max temp would be: 20.0
min temp would be 19.0
but as you can see from the automation topic its:
max temp = 20.5
min temp = 19.1

Am I missing something? How do these numbers calculate?
Really would love to understand this better as this is a really awesome project

**Comment by revk** [comment]
There is a setting called push which adjusts the targets, you can set that to 0.

**Reply by turbolooser** [reply]
Thanks @revk 
Giving it a try with "push":0

**Reply by turbolooser** [reply]
seems like I am geting an error when trying to send the push:0
I am pushing it to the topic "setting/faikin-schlafzimmer"
{"ts":"2024-02-03T10:55:01Z","error":"Not found","location":"\"push\":0}"}

**Reply by turbolooser** [reply]
I assume the parameter is called "pushtemp" and not "push" correct?

**Reply by revk** [reply]
Changed in beta, yes

**Reply by turbolooser** [reply]
much better than before but still some unknowns on the max temp (target: 17.5 )
{"hot":true,"approaching":0,"beyond":0,"samples":600,"period":300,"temp":17.93,"min":17.00,"max":18.50}
why is the max temp 0.5 higher than expected?

**Comment by turbolooser** [comment]
How to avoid/configure the command "set-fan":2 in the automations?
I do see this set-fan command once in a while ... is it trying to keep up with target temp? how to manage/configure?

just to add:
I am trying to run fanspeeds based on room presence. So if noone is in the room fanspeed could be high if required but if someone enters the room it should go silent again (mode:1,2,night).

so which settings for fanspeed would be optimal on non-presence and which on presence mode?
How to avoit to much of interference with the automation?

**Comment by turbolooser** [comment]
Sorry for having all these questions ...
why is the default for this thermref=50 ?

What do you mean by inlet? and what do you mean by Home?
how does this effect my BLE temp setup if I set this to 100?

<img width="775" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/29430360-ad59-4ec3-846a-b39ca7244bf0">

**Reply by revk** [reply]
The inlet or home are temperatures that get reported from the aircon. I think in my case one is related to a wall control and one is where air comes back in from the room (ducted). You can bias what it uses as a reference for setting temperatures in that Daikin auto mode. When using the BLE it compares to the reference and sets high or low against that reference. On my Daikin there is a config for how it uses those temperatures which thermref should match ideally. That said, I never got it to reliably use the temp sensor in the wall controller panel.

**Reply by turbolooser** [reply]
Could you show these temperatures on the status messages beeing sent?
I would love to see the home and the inlet temps if possible somewhere

**Reply by turbolooser** [reply]
I just detected a new variable which is
"env": 18
Is that a temp?

**Reply by revk** [reply]
env is usually the external environment sensor.

We *do* show temps on the status web page.

<img width="393" alt="Screenshot 2024-02-07 at 10 06 12" src="https://github.com/revk/ESP32-Faikin/assets/996983/29547cd0-09f7-465a-b1dd-aaa5943c7a7e">

**Reply by turbolooser** [reply]
I do see it in the mqtt message and its updated but I do not see it on the webgui
<img width="476" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/83c43bf3-c93a-4bf9-ab1a-dbbc9509e0e8">

**Reply by revk** [reply]
Ah, possibly because it is from BLE in your case, so you see BLE... Is it the same value?

**Reply by turbolooser** [reply]
Yeah thats kind of the same temp.

**Comment by turbolooser** [comment]
Having a target set to 19 how could I limit the maxtemp to be 19.5 rather than 20 ?
In my logic a target of 19 should run between 18.5 and 19.5 if I set the automode to have these lower/upper boundaries right?
{"hot":true,"approaching":0,"beyond":0,"samples":1800,"period":900,"temp":19.70,"min":18.50,"max":20.00}

**Reply by revk** [reply]
It will be a setting, I'm back from holiday some time tomorrow and will try and go through all these issues.

**Reply by turbolooser** [reply]
No issues ... just questions and grateful feedback to your awesome work on this 👍 
I love your project

**Reply by turbolooser** [reply]
I might have found the parametr 🥇 
{"switchtemp":0}
as my automation message now looks like this:
{"hot":true,"approaching":0,"beyond":605,"samples":1800,"period":900,"temp":19.67,"min":18.50,"max":19.50}

**Reply by revk** [reply]
That'll be it. What will be useful is for me to write up the way it works...

The various tweaks and settings were worked out over several months, not simple PID because I am fighting whatever logic the Daikin itself does. Ie if I just set the temperature I want, it will vary by quite a lot around that because that is what it does. So for tighter control (which is what I personally want) I have to make it switch on and off. I also found it would settle higher or lower than I want depending on heating or cooling, hence things like push temp and switch temp to move the goal posts slightly.

I hope that makes some sense.

**Reply by turbolooser** [reply]
Makes sense but on/off operations are not effective for at least multi-split AC units. Lowering the demand would be more efficient so that the units will continue to run (rather than off) but with the lowest amount of energy possible to keep the temperatures on target.

**Reply by turbolooser** [reply]
Just wanna share that it seems that I have found my sweet spot for my complex livingroom/wohnzimmer heating.

Maybe these parameters help others also (suggestions are also welcome)

autot = 19.5 (target faikin BLE temperature)
autor = 0.2 (margin to add to max temp and min temp. so the temp range in this case will be 19.3 up to 19.7 )(default = 0.5)
pushtemp = 0.3 (this pushes the min temp + 0.3 celsius (temp range is now 19.6 up to 19.7)(default = 0.1)
switchtemp = 0 (this pushes the max temp + 0 celsius) (temp range is now 19.6 up to 19.7)(default = 0.5)
tmin = 10 (perferas do have a min temp of 10 C) (default = 16)
tmax = 31 (perferas do have a max temp of 31 C) (default = 32)
heatback = 4 (set the faikin target temp to daikin-actual-temp minus 4 C) --> perferas behave little ugly in this
heatover = 5 (set the faikin target temp to daikin-actual-temp plus 5 C)
tsample = 600 (sample time of 10 mins rather than 15 mins)

<img width="998" alt="image" src="https://github.com/revk/ESP32-Faikin/assets/115738363/5db97628-d7fa-483b-bf66-97bda99cb8b9">

**Reply by MUN0X** [reply]
> Makes sense but on/off operations are not effective for at least multi-split AC units. Lowering the demand would be more efficient so that the units will continue to run (rather than off) but with the lowest amount of energy possible to keep the temperatures on target.

Thats what I’m trying to do with my multisplit. I use the thermostat function that was introduced a few weeks ago and use demand to keep it in range as long as possible. Minimizing the amount of on/off cycles.

**Reply by turbolooser** [reply]
@MUN0X 
Yes same here. Running it on demand 50/55 right now. As long as Liquid temps are high enough to give some heat to the room it should be fine. Getting here

**Comment by thomaslunze** [comment]
Is there any way to disable the fan-automation in auto mode? Or is there nothing which automaically adjusts the fan speeds?

**Reply by revk** [reply]
Not at present, it normally does full fan at the start (there is a setting for that), but adjust fan if heating or cooling seems to not be getting there, and then slows fan when it is. Which do you want to disable, or just all of it? I could add a setting.

**Reply by thomaslunze** [reply]
I guess all if possible. I am trying to run the fan settings on my own with some HA automations including presence detection and so on... so any fan speed automation would be unnecessary for me at least

If not all then maybe the dynamic ones while the unit is running. not sure if its like that

**Reply by revk** [reply]
OK I've added a `nofanauto` - try latest beta.

**Reply by thomaslunze** [reply]
awesome thanks a ton... really loving all ur efforts

**Comment by alex-nemirovsky** [comment]
@revk,
I finally had a chance to dig in and control the Faiken via MQTT. Thank you so much for creating this device. As I program it via MQTT, my appreciation for Faiken Auto Mode that you created has only grown.  It's very impressive!  I think I have things working that way I want, but I would like some clarification based on some of the posts in this thread.  Specifically  target min/max in relation to autot and autor.   My MQTT control messages are sent every minute containing an external room reference temperature via the env property in JSON.  While I am at it I also send other JSON properties like autot, autor, fanmode, target [min, max], mode, etc.. They way I currently understand things is that I send autot and autor to tell the Faiken that I would like the target temperature to be the autot value  +/- the autor value.   It tries to keep things in that range based on these two entries. Right?  Also, as I understand it the target[min, max] property is for auto power on and auto power off.  target[min, max]  NOT the values for ACTIVELY regulating temperature range but are only determining when the unit can go to sleep (OFF). i.e. if the temperature stays in this target region for 30 minutes without the need for the Faiken to adjust the temperature, then it goes into sleep (OFF) mode.  In sleep mode (OFF) the Faiken stops trying regulate the liquid temperature in the head unit.  It also, turns off the LED and closes the outlet flaps. If the temperature drops 0.1 C below the target[min] then the Faiken goes out of sleep (OFF) mode and into active stand by mode (ON). Where it evaluates and actively controls its internal liquid temperature (i.e. pressure)  and ask the heat pump adjust pressure as necessary to either heat or cool.. Similarly exceeding target[max] by 0.1 C then it also awakes from standby (OFF) and into active standby but this time switching to Cooling mode.   Is my understanding correct?   Is there any reason to send "mode" as H  or C in my MQTT control messages? I guess I am paranoid and that is what I am currently doing.  I dont want it to switch from heating to cooling by accident or vice versa..  I adjust the target[min, max]  based on room human presence but I adjust autot and autor at the same time.   Or I just totally confused and have this all wrong.

**Reply by revk** [reply]
There is a setting to stop it auto switching heating/cooling. Does that solve the problem?

**Reply by alex-nemirovsky** [reply]
could you elaborate on that setting which prevents mode switching?   also, is the rest of my understanding correct? I looking for confirmation or corrections.   I think the other heatpush and similar cooling control is only for when switching from sleep (OFF) to active (ON) mode.   To push the temperature beyond  the current temperatures in the hope that it can go back to sleep again (stop liquid temp management) and doesn't need to continue to stay in active (ON) mode.   When it is in active mode (ON) mode it doesn't use these values to manage the active target temperature range.   It's only a one time temperature adjust when coming out of sleep (OFF) mode.  right?

**Reply by revk** [reply]
I can look a bit more later, but the manual does have a lot of detail on the auto mode and options now, explaining the prediction and so on.
The settings is `lockmode` *Do not auto switch heat/cool modes in Faikin auto*

**Reply by alex-nemirovsky** [reply]
thanks. I think you are saying that my understanding is wrong.  :-)

**Reply by revk** [reply]
Not quite, I am saying I read the bit about not switching by accident, and I have only just got up and am now at coffee in town. I would suggest reading the manual to be sure, sorry. And if that does not help, I can try and go through in more details.

**Reply by alex-nemirovsky** [reply]
I definitely read the documentation. ;-)  several times over.  I found the lockmode, described in the WebUI comments. Looks it it prevents the user from turning off Faiken Auto mode from what I can tell.   I think I may need to keep forcing the mode to Heat or Cooling.    For example, it's winter here.    Faiken Auto mode seems pulls the liquid temperature target value way down below current env temperature when it determines it wants to tell the pump to stop applying pressure.  I guess I am paranoid that the pump will switch modes from heating to cooling when it sees the very low liquid temperature  target request.

**Reply by revk** [reply]
We only get to set mode and target temperature. How the Daikin then handles that is always fun. The liquid temp is just monitored and reported, not used in the Faikin code in any other way. Not sure if that helps explain.

My Daikin units work well and in spring/autumn can change heat cool during the day, usually only once. What happens if it starts one way, then when not needed it turns off, then it comes on the other way, as temp changes during the day.

But I keep my room within 0.5C target as I am fussy.

**Reply by alex-nemirovsky** [reply]
very helpful.  sounds like I need to observe it more.  thank you
